content/news/2019-12-03-mapstruct-and-quarkus.md

```diff
@@ -0,0 +1,220 @@
+
+{{< prettify >}}
+mvn package -Pnative
+{{< /prettify >}}
+
+_(In case you get an error that tests fail remove those or update them, I skipped this here as well)_
+
+As a prerequisite you need to have the GraalVM installed correctly. In case you don’t want or can install it on your machine take a look at the multi-stage Docker build in the previously mentioned guide.
+
+The command will produce an executable application in the `target` folder (it ends with `-runner`) that you can just start.
+
+You will notice that running the native compilation will take a while as it will not just compile your application as usual, it will afterwards create the native image. This process takes some time, but you will save this time afterwards in production when you need to (re)start your application plenty of times.
+
+The output after starting the app should look like this:
+
+{{< prettify >}}
+mapstruct-examples-quarkus 1.0-SNAPSHOT (running on Quarkus 1.0.1.Final) started in 0.006s. Listening on: http://0.0.0.0:8080
+Profile prod activated. 
+Installed features: [cdi, resteasy, resteasy-jsonb]
+{{< /prettify >}}
+
+I will repeat: _started in **0.006s**_.
+
+**This** is incredibly fast!
+
+### Summing up
+This example shows how easily it is to integrate MapStruct in a Quarkus application. This is a really huge plus-point for using MapStruct as the mapping framework in a Quarkus application compared to most of the other mapping frameworks that are using reflection.
+
+Therefore generating the mapper implementation does not only lead to be a [really fast](https://www.baeldung.com/java-performance-mapping-frameworks) mapping framework, but also to be a “native image” ready one!
+
+So yes: **MapStruct and Quarkus is definitely a match made in heaven!**
+
+The important point is to use CDI. The default component model (that’s where you retrieve the mapper with `Mappers.getMapper(PersonMapper.class)`) will not work out-of-the-box as this is the only part in MapStruct where reflection is used. If you still prefer this way you have to provide an own [reflection configuration](https://github.com/oracle/graal/blob/master/substratevm/REFLECTION.md). Maybe in the future MapStruct will also support this out-of-the box. But with version 1.3.1.Final it is not yet supported.
+
+
+#### One small disadvantage still left
+In case you are an experienced MapStruct user you might noticed that [the way how I added MapStruct](#adding-mapstruct) is a bit different to the way it is written in our [reference guide](http://mapstruct.org/documentation/stable/reference/html/#_apache_maven).
+
+The Quarkus development mode provides a hot-reload of your application, unfortunately it seems that the annotation processor will not be triggered again when you define the MapStruct processor within the annotation-processor path of the `maven-compiler-plugin` and you have to restart the application to see changes on the mappings.
```

Comment:
Could you log an issue in Quarkus for that one?

Reply:
You already did this :) See the reference in the paragraph afterwards (quarkusio/quarkus#1502)